### PR TITLE
update qindex when clicking a colony

### PR DESCRIPTION
**Pull request by skymandr**
QIndex is now updated when a colony is selected by clicking on the plate, magic interactions a little bit less magical.

### CODE REVIEW

File: scanomatic/ui_server_data/js/qc_normHelper.js [hunk 1]
```diff
@@ -293,7 +300,7 @@ function markExperiment(mark, all) {
     GetMarkExperiment(path, lockKey, (gData) => {
         if (gData.success === true) {
             dispatch.reDrawExp(`id${row}_${col}`, mark);
-            const queueCurrent = getQIndexCoord(qIdxOperations.Current);
+            const queueCurrent = updateQIndexCoord(qIdxOperations.Current);
```
[the remark this answers is not in the document]
Comment (skymandr) [reply scanomatic/ui_server_data/js/qc_normHelper.js]: umm... say what?

File: scanomatic/ui_server_data/js/qc_normHelper.js [hunk 2]
```diff
@@ -161,11 +162,8 @@ function fillProjectDetails(projectDetails) {
     });
 }
 
-function setExperimentByQidx(operation) {
-    const queueCurrent = getQIndexCoord(operation);
-    const row = queueCurrent.row;
-    const col = queueCurrent.col;
-    dispatch.setExp(`id${row}_${col}`);
+function getQIndexFromCoord(row, col) {
+    return qIndexQueue.filter(e => e.row == row && e.col == col)[0].idx;
```
Comment (local-minimum) [inline scanomatic/ui_server_data/js/qc_normHelper.js]: `===`

File: scanomatic/ui_server_data/js/qc_normHelper.js [hunk 3]
```diff
@@ -180,9 +178,11 @@ function updateQIndexLabel(qIndex) {
     $('#qIndexCurrent').text(qIndex + 1);
 }
 
-function getQIndexCoord(operation) {
+function updateQIndexCoord(operation, index) {
     if (operation === qIdxOperations.Reset) {
         qIndexCurrent = 0;
+    } else if (operation == qIdxOperations.Goto) {
```
Comment (local-minimum) [inline scanomatic/ui_server_data/js/qc_normHelper.js]: `===`

File: scanomatic/ui_server_data/js/qc_normHelper.js (hunk 2 above)
Comment (skymandr) [reply scanomatic/ui_server_data/js/qc_normHelper.js]: This made everything fail. My guess is that `e.row` is `str` and `row` is `int` etc. Reverted.

Comment (local-minimum) [reply scanomatic/ui_server_data/js/qc_normHelper.js]: I tested and it seems to be OK up to 48 at least for str == int and int == str.

**Pull request merged**